lib/block-supports/border.php

```diff
@@ -94,14 +94,18 @@ function gutenberg_apply_border_support( $block_type, $block_attributes ) {
 		! wp_should_skip_block_supports_serialization( $block_type, '__experimentalBorder', 'color' )
 	) {
 		$preset_border_color          = array_key_exists( 'borderColor', $block_attributes ) ? "var:preset|color|{$block_attributes['borderColor']}" : null;
-		$custom_border_color          = _wp_array_get( $block_attributes, array( 'style', 'border', 'color' ), null );
+		$custom_border_color          = isset( $block_attributes['style']['border']['color'] )
+			? _wp_array_get( $block_attributes, array( 'style', 'border', 'color' ), null )
```

Comment:
Can't this just use the following instead? What is realistically the difference?
```suggestion
			? $block_attributes['style']['border']['color']
```

Reply:
Good catch! I applied this tweak in all instances, it's a lot better now.
Thanks @felixarntz! 👍